core/src/main/scala/kafka/tools/StorageTool.scala

```diff
@@ -214,6 +340,24 @@ object StorageTool extends Logging {
     }
   }
 
+  def buildBootstrapMetadata(metadataVersion: MetadataVersion,
+                             metadataOptionalArguments: Option[ArrayBuffer[ApiMessageAndVersion]],
+                             source: String): BootstrapMetadata = {
+
+    val metadataRecords = new util.ArrayList[ApiMessageAndVersion]
+    metadataRecords.add(new ApiMessageAndVersion(new FeatureLevelRecord().
+                        setName(MetadataVersion.FEATURE_NAME).
+                        setFeatureLevel(metadataVersion.featureLevel()), 0.toShort));
+
+    metadataOptionalArguments.foreach { metadataArguments =>
+      for (record <- metadataArguments) metadataRecords.add(record)
+    }
+
+    metadataRecords.add(new ApiMessageAndVersion(new NoOpRecord(), 0.toShort));
```

Comment:
why are you adding a no-op record?

Reply:
Not sure :) I must have added it early when I was POC that I could make the bootstrap code work for SCRAM and then later thought it was always there. I'll remove it.

Reply:
Fixed